src/plugins/es_ui_shared/public/index.ts

```diff
@@ -68,6 +68,8 @@ export {
 
 export { Monaco, Forms };
 
+export { extractQueryParams } from './url';
```

Comment:
@cjcenizal @jloleysens I think we need to write some guidelines for our team about how to export from "es_ui_shared". I _thought_ that we were aligned but I see we are not, and that's probably my mistake of not spreading the word.

The guidelines to follow, like we can see on L71, is to export **namespace packages** from the es_ui_shared folder. This implies to group items by "domain". Here I would have suggested having a "Routing" domain. 

Each domain represents a top-level folder that lives inside the "__packages_do_not_import__" folder.

My main concerns are:

* name collision. I don't want to have to tweak my interface names to not collide with other interfaces from other existing domains. If `UpdateHandler` makes sense in my context, I want to be able to use it, even if another domain is using it in a different context.

* Pollute the public `index.ts` with all the interfaces, constants, functions from all the packages. This file will keep growing with time and it's much better to export "packages" like we do with

```
export { Monaco, Forms, ace };
```

If we need to know what a package contains, we then go and look at its barrel file inside its folder.

Until the guidelines are written I think it is important to keep an eye on that during code reviews. WDYT?

Reply:
@sebelga , I do agree with this overall. I think it is best in as far as possible to follow the convention you've outlined (which is why I created https://github.com/elastic/kibana/pull/72034 in response to your comment :) ).

In this case naming of the function seemed so specific that I was not worried collision protection. That being said it would probably have been best if I'd also called it out here.

The other aspect of putting something that is fairly uniquely named inside of a module like `Routing.extractQueryParams` is the added step of unwrapping it later or updating all uses to have the module name prefix. On my initial review I just thought it might be overkill.

But again, in retrospect, I think it should have been called it attention here.

Reply:
I was not pointing you directly about the review 😊 Just sharing thoughts/concerns. Thanks for creating #72034 !

The thing about thinking "_naming of the function seemed so specific that I was not worried collision protection_" is that it is subject to interpretation so I think it is better to follow a pattern and stick to it.

For the extra unwrapping, yes there is an extra step but I think it is worth in the long run. True that we might need to update the code in a few places but I didn't find it too complex to do when I worked with Forms

```js
// instead of this
import { FormWizard } from '../../shared_imports';

// we have this
import { Forms } from '../../shared_imports';
const { FormWizard } = Forms;
```

IMO it makes it very clear what reusable package(s) a file uses and we can understand the context of a specific handler/interface.